include/picongpu/particles/atomicPhysics/SetToAtomicGroundStateForChargeState.hpp

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2021 Brian Marre, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <pmacc/traits/HasFlag.hpp>
+
+namespace picongpu
+{
+    namespace particles
+    {
+        namespace atomicPhysics
+        {
+            struct SetToAtomicGroundStateForChargeState
+            {
+                template<typename T_Particle>
+                DINLINE void operator()(T_Particle& particle, uint8_t numberBoundElectrons)
```

Comment:
please use `float_X` instead of `uint8_t` everywhere. The attribute `boundElectrons` is `float_X` too and we should avoid implicit casts everywhere.

Reply:
Please check your static casts everywhere where you are calling `SetToAtomicGroundStateForChargeState`